scripts/lint_warn_assert.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -euo pipefail
+#
+# lint_warn_assert.sh - Shared helper for testify/assert advisory warnings
+#
+# Usage:
+#   source /path/to/lint_warn_assert.sh
+#   test_warn_testify_assert
+
+# Advisory check for testify/assert usage - warns but doesn't fail.
+# Only runs on CI PRs, checking new code only.
+# assert continues execution after failure, require fails fast.
+# Developers should consciously choose when assert is appropriate.
```

Comment:
(no action required) @ARR4N something on when to use each in matters of style for this could be cool!